spec/controllers/surveys_controller_spec.rb

```diff
@@ -253,4 +253,95 @@
       expect(Survey.last.parameters).not_to include('unexpected')
     end
   end
+
+  describe 'GET #index' do
+    let(:first_survey_parameters) {
+      {
+        name: 'My first survey',
+        description: 'This is my first survey',
+        questions: [
+          {
+            id: 1,
+            text: 'What is your name?',
+            question_type: 'short_answer',
+            default_next_question_id: 2,
+            options: [],
+          }
+        ]
+      }
+    }
+
+    let(:second_survey_parameters) {
+      {
+        name: 'Second survey',
+        description: 'This is my second survey',
+        questions: []
+      }
+    }
+
+    let(:first_expected_survey) {
+      {
+        id: 1,
+        name: 'My first survey',
+        description: 'This is my first survey',
+        number_of_questions: 1
+      }
+    }
+
+    let(:second_expected_survey) {
+      {
+        id: 2,
+        name: 'Second survey',
+        description: 'This is my second survey',
+        number_of_questions: 0
+      }
+    }
+
+    context 'no surveys' do
+      it 'returns http success' do
+        get :index
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'returns empty response' do
+        get :index
+        expect(response.body).to eq({ surveys: [] }.to_json)
+      end
+    end
+
+    context 'single survey' do
+      it 'returns http success' do
+        FactoryGirl.create(:survey, id: 1, parameters: first_survey_parameters.to_json)
```

Comment:
You can move this to line 313 and it will be executed at context scope so all tests within the context get it. Then you don't have to repeat it on line 321.